lib/middleware/authenticate.js

```diff
@@ -1,9 +1,19 @@
 const jwt = require('jsonwebtoken');
 
 module.exports = async (req, res, next) => {
-  /*
-    TODO: Check for the session cookie and verify
-    its contents using jsonwebtoken, then
-    assign the payload to req.user
-  */
+  try {
+    //TODO: Check for the session cookie and
+    const cookie = req.cookies && req.cookies[process.env.COOKIE_NAME];
+    // verify
+    if (!cookie) throw new Error('You must be signed in to continue');
+    //its contents using jsonwebtoken,
+    const user = jwt.verify(cookie, process.env.JWT_SECRET);
+    // then assign the payload to req.user
```

Comment:
i like how you broke up the comment steps to match the code you needed, that's a good way to think about it